<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>642</width>
    <height>463</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HardwareFullTest</string>
  </property>
  <property name="windowIcon">
   <iconset resource="rs.qrc">
    <normaloff>:/Chip.png</normaloff>:/Chip.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Serial</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="1">
         <widget class="QLabel" name="label_7">
          <property name="font">
           <font>
            <family>Consolas</family>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>Serial Ports</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <spacer name="horizontalSpacer_2">
          <property name="styleSheet">
           <string>background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Minimum</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>160</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0">
         <spacer name="horizontalSpacer">
          <property name="styleSheet">
           <string>background-color: rgb(255, 255, 255);</string>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Minimum</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>160</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="1">
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QComboBox" name="cb_serial">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cb_baudrate">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="bt_open">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>12</pointsize>
              <weight>50</weight>
              <bold>false</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>Open Serial</string>
            </property>
            <property name="autoDefault">
             <bool>false</bool>
            </property>
            <property name="default">
             <bool>true</bool>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="bt_close">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>Close Serial</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="refresh_bt">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>Refresh</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="bt_clear">
            <property name="font">
             <font>
              <family>Consolas</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="1">
         <widget class="QPlainTextEdit" name="txt_receber">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_15">
       <attribute name="title">
        <string>Camera</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>IMU</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Yaw</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="tb_yaw_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Pitch</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="tb_pitch_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Roll</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="tb_roll_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="bt_setRot_2">
            <property name="text">
             <string>Set Rotation</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="bt_calibrate">
            <property name="text">
             <string>Calibrate</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_2">
            <property name="text">
             <string>Quit</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="GLWidget" name="widget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>GamePad Control</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="8" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="lb_horizontal">
            <property name="text">
             <string>PressedH:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lb_vertical">
            <property name="text">
             <string>PressedV:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>TextLabel</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="serialLB">
            <property name="text">
             <string>TextLabel</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="7" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QProgressBar" name="LjoyX">
            <property name="minimum">
             <number>-32767</number>
            </property>
            <property name="maximum">
             <number>32767</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="format">
             <string>%v</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="LjoyY">
            <property name="minimum">
             <number>-32767</number>
            </property>
            <property name="maximum">
             <number>32767</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="format">
             <string>%v</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="RjoyX">
            <property name="minimum">
             <number>-32767</number>
            </property>
            <property name="maximum">
             <number>32767</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="format">
             <string>%v</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="RjoyY">
            <property name="minimum">
             <number>-32767</number>
            </property>
            <property name="maximum">
             <number>32767</number>
            </property>
            <property name="value">
             <number>0</number>
            </property>
            <property name="format">
             <string>%v</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="0">
         <widget class="QGraphicsView" name="graphicsView">
          <property name="styleSheet">
           <string notr="true">background-image: url(:/Gamepad.png);
background-repeat: no-repeat;
background-position: center;
</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Virtual Joystick</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QGraphicsView" name="graphicsViewJoy">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>350</width>
            <height>360</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">border-image: url(:/joy.png) 0 0 0 0 stretch stretch;
border-width: 1px black solid;</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="lb_values">
            <property name="text">
             <string>0;0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="1" rowspan="2">
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QSlider" name="verticalSliderKick">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBelow</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lb_kick">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>0%</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="bt_kickV">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Vertical Kick</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="bt_kickH">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Horizontal Kick</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Arduino Code</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="1" column="0">
         <widget class="QPushButton" name="bt_copiar">
          <property name="text">
           <string>Gerar Codigo</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QTabWidget" name="tabCode">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tab_6">
           <attribute name="title">
            <string>Arduino V1</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <widget class="QPlainTextEdit" name="codeArduino">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_10">
           <attribute name="title">
            <string>Ir</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_10">
            <item row="0" column="0">
             <widget class="QPlainTextEdit" name="codeIr">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_16">
       <attribute name="title">
        <string>Bibiliotecas Tutorial</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_15">
        <item row="0" column="0">
         <widget class="QTabWidget" name="tabWidget_2">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <widget class="QWidget" name="tab_17">
           <attribute name="title">
            <string>Windows</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_16">
            <item row="0" column="0">
             <widget class="QPlainTextEdit" name="plainTextEdit">
              <property name="readOnly">
               <bool>true</bool>
              </property>
              <property name="plainText">
               <string>Neste Software são usadas duas Bibiolitecas (extra QT 5.3.2).

São elas:

SDL -Simple DirectMedia Layer é uma biblioteca multimídia livre e de código aberto, multiplataforma, escrito em C que representa uma interface simples para gráficos, som, e dispositivos de entrada de várias plataformas. (Usada para o reconhecimento de qualquer GAMEPAD)
 
GLUT-GLUT (OpenGL Utility Toolkit) é uma biblioteca de funcionalidades para OpenGL cujo principal objetivo é a abstração do sistema operacional fazendo com que os aplicativos sejam multiplataforma. (Usada para desenhar em 3D o robô para melhor visualização dos dados adiquiridos do IMU)

Para que funcionem correctamente com o compilador do WINDOWS é necessário que estejam declaradas no .pro deste projecto e que estejam disponiveis no seguinte caminho: 

LIBS += -L&quot;C:\Users\Helder\Documents\QtProjetcts\GLUT&quot; \
        -lglut32

LIBS += -L&quot;C:\Users\Helder\Documents\QtProjetcts\SDL\x86&quot; \
        -lSDL2

(Sendo este um caminho absoluto, é necessário ser alterado sempre que o codigo seja compilado num outro PC)

Ambas as bibliotecas podem ser encontradas online, ou na pasta de instalação deste programa em formato ZIP.

Para além disto é necessário incluir os .h das respectivas bibliotecas, (que se encontram nesse mesmo ZIP dentro da pasta &quot;include&quot;), devem então ser adicionados ao compilador em uso, por exemplo em: &quot;C:\Qt\5.3\mingw482_32\include&quot;


</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_18">
           <attribute name="title">
            <string>Linux</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0">
             <widget class="QPlainTextEdit" name="plainTextEdit_2">
              <property name="plainText">
               <string>Neste Software são usadas duas Bibiolitecas (extra QT 5.3.2).

São elas:

SDL -Simple DirectMedia Layer é uma biblioteca multimídia livre e de código aberto, multiplataforma, escrito em C que representa uma interface simples para gráficos, som, e dispositivos de entrada de várias plataformas. (Usada para o reconhecimento de qualquer GAMEPAD)
 
GLUT-GLUT (OpenGL Utility Toolkit) é uma biblioteca de funcionalidades para OpenGL cujo principal objetivo é a abstração do sistema operacional fazendo com que os aplicativos sejam multiplataforma. (Usada para desenhar em 3D o robô para melhor visualização dos dados adiquiridos do IMU)

Para que funcionem correctamente com o Ubuntu é necessário que estejam declaradas no .pro deste projecto e que estejam disponiveis no seguinte caminho: 

LIBS += -L&quot;/home/code/Documents/LIBS/SDL&quot; \
        -lSDL2

LIBS += -L&quot;/home/code/Documents/LIBS/GLUT&quot; \
        -lglut -lGLU -lGL

(Sendo este um caminho absoluto, é necessário ser alterado sempre que o codigo seja compilado num outro PC)

Ambas as bibliotecas podem ser encontradas online, ou na pasta de instalação deste programa em formato tar.gz.




</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="rs.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>114</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>448</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
